packages/core/src/types.ts

```diff
@@ -41,7 +41,7 @@ export type GetParameterizedParams<T extends ParameterizedObject | undefined> =
  * This type can be used to avoid this problem. This union represents the same
  * value space as `unknown`.
  */
-export type NonReducibleUnknown = {} | null | undefined;
+export type NonReducibleUnknown = NonNullable<unknown> | null | undefined;
```

Comment:
I officially hate this... :joy: I kinda understand why `{}` has such a bad rep... but if I'll have to start using `NonNullable<unknown>` everywhere to workaround a lint rule... 😬 

Personally, I'd vote for disabling that rule and allowing `{}`